### ISSUE TITLE
`ExprKind::Spawn` has no producer — the block form `spawn { }` is in SYNTAX.md, isn't the language, and carries ~60 lines of unreachable interpreter code

### ISSUE THREAD

**Issue by dritory** [rask-lang/rask#1115]
`spawn { body }` doesn't parse:

```rask
let h = spawn { println("hi") }
```

```
error[E0100]: Expected ';' or newline after statement
  --> sp.rk:5:23
    |
  5 |         let h = spawn { println("hi") }
    |                       ^ here
```

Nothing in the parser builds `ExprKind::Spawn`. Grepping every crate for a construction site finds exactly one, and it's a unit-test fixture (`rask-effects/src/infer.rs:954`). Everything else is a consumer: ten `match` arms across `rask-ownership`, `rask-fmt`, `rask-lsp`, `rask-mir` (four), `rask-mono`, `rask-effects`, `rask-desugar` (two), `rask-interp` and `rask-ast`'s own walker.

The real syntax is `spawn(|| { … })`, which parses as a `Call` and resolves to `BuiltinKind::AsyncSpawn`.

## What that costs

The interpreter's `ExprKind::Spawn` arm (`eval_expr.rs:2830`) is ~60 lines that build a `PoolTask` and submit it to the `using Multitasking(n)` worker pool. It is unreachable. The reachable path — `call_async_method` — starts an OS thread per task and never touches the pool.

That's what made #1111 read as "there are two spawn implementations, and the reachable one ignores the pool". There is one implementation and one corpse, which is a different problem with a different fix. Correcting that there.

## The state is confused in four places

- **`specs/SYNTAX.md`** shows the block form three times (`spawn { handle(conn) }.detach()`, `let handle = spawn { compute() }`, `spawn { background_work() }.detach()`), and `specs/stdlib/net.md` and `specs/stdlib/README.md` show it once each.
- **`CLAUDE.md`'s Decided table** says the opposite: "Concurrency | `spawn(|| {})`/join/detach (functions)". `specs/concurrency/` agrees with the table.
- **`specs/compiler/codegen.md`** lists "Green task runtime — `spawn { }` requires M:N scheduler. Defer until after basic threading" as deferred work, which is a legitimate reason for a placeholder to exist.
- **`tests/suite/t_month_concurrency.rk`** explains the situation in a comment and cites **#267** for it. #267 is closed, and was about registering `spawn(|| {})` as a builtin — a different thing. Nothing open tracks the block form.

## The decision

Either green tasks are still coming, in which case the placeholder is fine and SYNTAX.md is describing a planned form (it should say so, and this issue is what the test comment should cite); or `spawn(|| {})` is the whole story, in which case the variant, its ten arms and the unreachable interpreter path go, and SYNTAX.md's three examples change.

I'd take the second — the Decided table already settles the surface syntax, and a placeholder with no producer that nothing tracks is the shape CLAUDE.md calls rot with a delay fuse. But it is a roadmap call about green tasks, not a cleanup, so it isn't mine.

Found while investigating #1111.
